src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -49,27 +53,27 @@ public void initialize(){
     public void execute(){
         shooterSpeedController.setLaunchSpeed(ShooterMath.determineLaunchSpeed(distanceToGoal + INNER_DISTANCE_FROM_TARGET));
 
-        for(int ballsShot = 0 ; ballsShot < 3 ; ballsShot ++){
-         // Make sure the shooter is at speed before loading a power cell
-         if (shooterSpeedController.isAtSpeed()) {
-             System.out.println("Shot Balls" + ballsShot);
-            // turn on the indexer and hopper
-            if(hasBall){
-                this.indexerController.intake();
-                this.hopperController.stop();
+        while(ballsShot < 3){
+            while(ballShot == false){
```

Comment:
you don't need a while loop here either.  Think of execute as something that happens once each time through a loop.  We don't want anything stopping that loop from running all of the time.  Instead I would make this a series of ifs and have no whiles at all.

Remember, we're going to:
turn the shooter on for the distance we're at.  You do that in initialize.

Then each time through the loop (in execute())
check if the shooter is at speed and there's no ball in progress and (Timer.getFPGATimeStamp() - startTime) > COOLDOWN TIMER
     if so, turn on the indexer and hopper, set launch in progress to true
check if the shooter is at speed and launch in progress is true
  check if a ball is detected (call distanceSensorDetector.isBallDetected()) and ballInIndexer is false
     if so, set ballInIndexer as true
  check if a ball is not detected and ballInIndexer is true
    turn hopper and indexer off
    ++ballShot
    set ballInIndexer to false
    set launchInProgress to false
    set startTime to Timer.getFPGATimestamp()

and that's it.  Maybe set COOLDOWN_TIMER to 0.5?  I have no idea how long we'll need to cooldown, it might be 0

Reply:
I wrote the code, but I don't see why launch in progress is necessary in the newly updated code

Reply:
when you have time can you please look at the current code and critic it?

Reply:
@elizaBarnett I made another review.  You are completely right, we don't need launch in progress